### PR TITLE
[12.0][MIG] l10n_nl_bsn

### PR DESCRIPTION
**Pull request by astirpe**

### CODE REVIEW

**Review by espo-tony — APPROVED** [review]
LGTM :+1:

File: l10n_nl_bsn/README.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+   :target: https://www.gnu.org/licenses/agpl
+   :alt: License: AGPL-3
+
+======================================
+Burgerservicenummer (BSN) for Partners
+======================================
+
+This module adds the BSN (Burgerservicenummer) field on partner forms.
+
+The field is visible when the flag is_company is false.
```
Comment (espo-tony) [inline l10n_nl_bsn/README.rst]: This line is not true anymore, as flag is_company was removed in v12. The field is visible, instead, when "Individual" radio button is checked.

File: l10n_nl_bsn/README.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+* User must belong to the "HR Officer" group
+
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Open a form of a contact, eg.: a person (uncheck the flag "Is a Company?")
```
Comment (espo-tony) [inline l10n_nl_bsn/README.rst]: As above, this line is not accurate anymore

File: l10n_nl_bsn/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            if partner.bsn_number:
+                # properly format the entered BSN
+                partner.bsn_number = bsn.format(partner.bsn_number)
+
+                # check is valid, otherwise display a warning
+                warning = partner._warn_bsn_invalid()
+
+                # search for another partner with the same BSN
+                args = [('bsn_number', '=', partner.bsn_number),
+                        ('name', '!=', partner.name)]
```
Comment (espo-tony) [inline l10n_nl_bsn/models/res_partner.py:37]: Why did you add a check on the name within the search domain?

File: l10n_nl_bsn/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+                # refine search in case of multicompany setting
+                if partner.company_id:
+                    args += [('company_id', '=', partner.company_id.id)]
+                other_partner = self.search(args, limit=1)
+
+                # if another partner exists, display a warning
+                if other_partner:
+                    warning = other_partner._warn_bsn_existing()
+        return {'warning': warning, }
```
Comment (espo-tony) [inline l10n_nl_bsn/models/res_partner.py:47]: If the BSN is not well formatted AND is already in use, you'd get a warning only for the latter, as the former value of the variable would be overwritten.
Is that the wished behavior? Wouldn't be better to have both issues signaled at once?

**Review by espo-tony — CHANGES_REQUESTED** [review]
I tested the module and it works fine. Still I have some lesser changes to request and some questions.

File: l10n_nl_bsn/README.rst (hunk 1 above)
Comment (astirpe) [reply l10n_nl_bsn/README.rst]: This is already fixed in file `readme/DESCRIPTION.rst` (https://github.com/OCA/l10n-netherlands/pull/199/files#diff-351c10adac39025dad8b1371e573db34R3). The `README.rst` will be automatically overwritten.

File: l10n_nl_bsn/README.rst (hunk 2 above)
Comment (astirpe) [reply l10n_nl_bsn/README.rst]: Done in https://github.com/OCA/l10n-netherlands/pull/199/files#diff-de53f2d9eb0ad4e03a14910110c926dfR3

File: l10n_nl_bsn/models/res_partner.py (hunk 3 above)
Comment (astirpe) [reply l10n_nl_bsn/models/res_partner.py:37]: Because you may have multiple records representing the same partner. For example, if 2 records of _PartnerA_ are present: you may want to accept both records of _PartnerA_  having the same `bsn_number`.

File: l10n_nl_bsn/models/res_partner.py (hunk 4 above)
Comment (astirpe) [reply l10n_nl_bsn/models/res_partner.py:47]: I assumed that the warning "already in use" has the priority over the "not well formatted" one. You would likely check the "already in use" case first, in case both conditions happen.

File: l10n_nl_bsn/models/res_partner.py (hunk 3 above)
Comment (astirpe) [reply l10n_nl_bsn/models/res_partner.py:37]: See also https://github.com/OCA/l10n-netherlands/pull/43

Comment (espo-tony) [reply l10n_nl_bsn/models/res_partner.py:37]: Ok. Thanks for your reply.

File: l10n_nl_bsn/models/res_partner.py (hunk 4 above)
Comment (espo-tony) [reply l10n_nl_bsn/models/res_partner.py:47]: If it's the wished behavior that's fine by me. Thanks

**Review by espo-tony — APPROVED** [review]
:+1:

**Review by PatrickMallee — APPROVED** [review]
works fine

File: l10n_nl_bsn/README.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.png
```
Comment (tarteo) [inline l10n_nl_bsn/README.rst]: Can you generate the README.

**Review by tarteo — CHANGES_REQUESTED** [review]

**Review by tarteo — APPROVED** [review]
Thanks!

**Comment by astirpe** [comment]
With latest commit I'm proposing myself as maintainer of this module
